src/utils/hooks/usePaginering.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { ReactNode, useEffect, useState } from 'react';
+import { Select } from 'nav-frontend-skjema';
+
+interface Returns<T> {
+    currentPage: T[];
+    pageSelect: ReactNode;
+}
+
+function usePaginering<T>(list: T[], pageSize: number): Returns<T> {
+    const [currentPage, setCurrentPage] = useState(0);
+
+    const numberOfPages = Math.ceil(list.length / pageSize);
+
+    useEffect(() => {
+        if (currentPage >= numberOfPages) {
+            setCurrentPage(0);
+        }
+    }, [list, numberOfPages, currentPage]);
+
+    if (list.length <= pageSize) {
+        return {
+            currentPage: list,
+            pageSelect: null
+        };
+    }
+
+    const getRange = (index: number) => {
+        return {
+            fra: index * pageSize,
+            til: (index + 1) * pageSize < list.length ? (index + 1) * pageSize - 1 : list.length - 1
+        };
+    };
+
+    const options = [...new Array(numberOfPages)].map((_, index) => {
```

Comment:
Meget personlig mening, men jeg liker nok `new Array(numberOfPages).fill(0).map(....)` syntaksen bedre enn spread-operatoren her.

Reply:
Mmm, da bytter jeg for din skyld ;-)